### ISSUE TITLE
Crash during destroy

### ISSUE THREAD

**Issue by zeuxis** [hashicorp/terraform#9514]
Hi!

terraform crashed when destroying an infrastructure:
### Terraform Version

v0.7.7, freshly compiled
### crash.log

can be downloaded from: https://acentrix.sharefile.com/d-see433c7884344bca
### Fix

rerunning `terraform destroy` did not fix the problem, the VPC `aws_vpc.shared` had to be deleted manually

**Comment by jbardin** [comment]
Thanks for the crash report!

The relevant traceback from the log

```
2016/10/21 19:11:03 [DEBUG] plugin: terraform: panic: runtime error: invalid memory address or nil pointer dereference
2016/10/21 19:11:03 [DEBUG] plugin: terraform: [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x9c48fc]
2016/10/21 19:11:03 [DEBUG] plugin: terraform:
2016/10/21 19:11:03 [DEBUG] plugin: terraform: goroutine 1732 [running]:
2016/10/21 19:11:03 [DEBUG] plugin: terraform: panic(0x28c59c0, 0xc42000e100)
2016/10/21 19:11:03 [DEBUG] plugin: terraform:  /opt/go/src/runtime/panic.go:500 +0x1a1
2016/10/21 19:11:03 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/builtin/providers/aws.deleteAllRecordsInHostedZoneId.func1(0xc4207eb680, 0x1, 0x0)
2016/10/21 19:11:03 [DEBUG] plugin: terraform:  /opt/gopath/src/github.com/hashicorp/terraform/builtin/providers/aws/resource_aws_route53_zone.go:327 +0x7bc
2016/10/21 19:11:03 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/vendor/github.com/aws/aws-sdk-go/service/route53.(*Route53).ListResourceRecordSetsPages.func1(0x291d940, 0xc4207eb680, 0x1, 0xc42064c020)
2016/10/21 19:11:03 [DEBUG] plugin: terraform:  /opt/gopath/src/github.com/hashicorp/terraform/vendor/github.com/aws/aws-sdk-go/service/route53/api.go:2236 +0x49
2016/10/21 19:11:03 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/vendor/github.com/aws/aws-sdk-go/aws/request.(*Request).EachPage(0xc420577880, 0xc420823700, 0x4, 0x4)
2016/10/21 19:11:03 [DEBUG] plugin: terraform:  /opt/gopath/src/github.com/hashicorp/terraform/vendor/github.com/aws/aws-sdk-go/aws/request/request_pagination.go:98 +0x90
2016/10/21 19:11:03 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/vendor/github.com/aws/aws-sdk-go/service/route53.(*Route53).ListResourceRecordSetsPages(0xc4200321a0, 0xc420756de0, 0xc4208237c0, 0x2713400, 0xc42045c2d0)
2016/10/21 19:11:03 [DEBUG] plugin: terraform:  /opt/gopath/src/github.com/hashicorp/terraform/vendor/github.com/aws/aws-sdk-go/service/route53/api.go:2237 +0x116
2016/10/21 19:11:03 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/builtin/providers/aws.deleteAllRecordsInHostedZoneId(0xc4206b3840, 0xe, 0xc4206b3b28, 0x8, 0xc4200321a0, 0xc42004a900, 0x418afe)
2016/10/21 19:11:03 [DEBUG] plugin: terraform:  /opt/gopath/src/github.com/hashicorp/terraform/builtin/providers/aws/resource_aws_route53_zone.go:333 +0x193
2016/10/21 19:11:03 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/builtin/providers/aws.resourceAwsRoute53ZoneDelete(0xc420758c00, 0x2623360, 0xc420456600, 0x7000000000059, 0x414f21)
2016/10/21 19:11:03 [DEBUG] plugin: terraform:  /opt/gopath/src/github.com/hashicorp/terraform/builtin/providers/aws/resource_aws_route53_zone.go:269 +0x19e
2016/10/21 19:11:03 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/helper/schema.(*Resource).Apply(0xc4202948a0, 0xc42077f590, 0xc4208c3dc0, 0x2623360, 0xc420456600, 0x1, 0xc4208baf60, 0x18)
2016/10/21 19:11:03 [DEBUG] plugin: terraform:  /opt/gopath/src/github.com/hashicorp/terraform/helper/schema/resource.go:137 +0x3eb
2016/10/21 19:11:03 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/helper/schema.(*Provider).Apply(0xc420445440, 0xc4207eaa40, 0xc42077f590, 0xc4208c3dc0, 0x0, 0xff595b68e1538eb7, 0xd5a79147c6e00bf3)
2016/10/21 19:11:03 [DEBUG] plugin: terraform:  /opt/gopath/src/github.com/hashicorp/terraform/helper/schema/provider.go:178 +0x9b
2016/10/21 19:11:03 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Apply(0xc4204403c0, 0xc4208c3c80, 0xc4206b3b40, 0x0, 0x0)
2016/10/21 19:11:03 [DEBUG] plugin: terraform:  /opt/gopath/src/github.com/hashicorp/terraform/plugin/resource_provider.go:460 +0x57
2016/10/21 19:11:03 [DEBUG] plugin: terraform: reflect.Value.call(0xc42045a3c0, 0xc4201d28a8, 0x13, 0x2e86ddb, 0x4, 0xc420823ed0, 0x3, 0x3, 0x26fdc00, 0xc420876a20, ...)
2016/10/21 19:11:03 [DEBUG] plugin: terraform:  /opt/go/src/reflect/value.go:434 +0x5c8
2016/10/21 19:11:03 [DEBUG] plugin: terraform: reflect.Value.Call(0xc42045a3c0, 0xc4201d28a8, 0x13, 0xc420823ed0, 0x3, 0x3, 0xc42039c014, 0x100000000, 0xc420513f01)
2016/10/21 19:11:03 [DEBUG] plugin: terraform:  /opt/go/src/reflect/value.go:302 +0xa4
2016/10/21 19:11:03 [DEBUG] plugin: terraform: net/rpc.(*service).call(0xc420725d40, 0xc420725d00, 0xc42045d848, 0xc4201ba880, 0xc42010e1c0, 0x2625220, 0xc4208c3c80, 0x16, 0x2625260, 0xc4206b3b40, ...)
2016/10/21 19:11:03 [DEBUG] plugin: terraform:  /opt/go/src/net/rpc/server.go:383 +0x148
2016/10/21 19:11:03 [DEBUG] plugin: terraform: created by net/rpc.(*Server).ServeCodec
2016/10/21 19:11:03 [DEBUG] plugin: terraform:  /opt/go/src/net/rpc/server.go:477 +0x421
```

**Comment by catsby** [comment]
This should be guarded / fixed by #10796 and #10798
